uint32 pm_id
uint32 update_counter
uint32[] events
time[] event_timestamps
